bashdb: fails to build on bash-5 (supports only bash-4)

`bashdb` fails build as  https://hydra.nixos.org/build/154344775:

```
configure: WARNING: You have Bash GNU bash, version 5.1.8(1)-release (x86_64-pc-linux-gnu)
...
configure: error: This package is only known to work with Bash 4.1-4.4.
```

Should we force package to `bash_4`, or maybe remove it?

### Notify maintainers

No maintainer. 

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: bashdb
```

--- a/pkgs/development/tools/misc/bashdb/default.nix
+++ b/pkgs/development/tools/misc/bashdb/default.nix
@@ -1,13 +1,29 @@
-{ lib, stdenv, fetchurl, makeWrapper, python3Packages }:
+{ lib
+, stdenv
+, fetchurl
+, fetchpatch
+, makeWrapper
+, python3Packages
+}:
 
 stdenv.mkDerivation rec {
   pname = "bashdb";
-  version = "4.4-1.0.0";
+  version = "5.0-1.1.2";
 
   src = fetchurl {
     url =  "mirror://sourceforge/bashdb/${pname}-${version}.tar.bz2";
-    sha256 = "0p7i7bpzs6q1i7swnkr89kxqgzr146xw8d2acmqwqbslzm9dqlml";
+    sha256 = "sha256-MBdtKtKMWwCy4tIcXqGu+PuvQKj52fcjxnxgUx87czA=";
   };
+
+  patches = [
+    # Enable building with bash 5.1/5.2
+    # Remove with any upstream 5.1-x.y.z release
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/freebsd/freebsd-ports/569fbb806d9ee813afa8b27d2098a44f93433922/devel/bashdb/files/patch-configure";
+      sha256 = "19zfzcnxavndyn6kfxp775kjcd0gigsm4y3bnh6fz5ilhnnbbbgr";
+    })
+  ];
+  patchFlags = "-p0";
 
   nativeBuildInputs = [
     makeWrapper

